hal/audio_extn/cirrus_sony.c

```diff
@@ -600,17 +594,48 @@ static int cirrus_play_silence(int seconds) {
     struct audio_device *adev = handle.adev_handle;
     struct mixer_ctl *ctl_config = NULL;
     struct pcm_config rx_tmp = { 0 };
+    struct audio_usecase *uc_info_rx;
+    #define CIRRUS_PLAYBACK_MIXER "SEN_MI2S_RX Audio Mixer MultiMedia1"
 
     uint8_t *silence = NULL;
-    int i, ret, silence_bytes, silence_cnt = 1;
+    int i, ret = 0, silence_bytes, silence_cnt = 1;
     unsigned int buffer_size = 0, frames_bytes = 0;
+    int pcm_dev_rx_id;
+
+    if (!list_empty(&adev->usecase_list)) {
+        ALOGD("%s: Usecase present retry speaker protection", __func__);
+        return -EAGAIN;
+    }
+
+    uc_info_rx = (struct audio_usecase *)calloc(1, sizeof(struct audio_usecase));
+    if (!uc_info_rx) {
+        return -ENOMEM;
+    }
+    uc_info_rx->id = USECASE_AUDIO_PLAYBACK_DEEP_BUFFER /* USECASE_AUDIO_SPKR_CALIB_RX */;
+    uc_info_rx->type = PCM_PLAYBACK;
+    uc_info_rx->in_snd_device = SND_DEVICE_NONE;
+    uc_info_rx->stream.out = adev->primary_output;
+    list_init(&uc_info_rx->device_list);
+    // if (fp_audio_extn_is_vbat_enabled())
+    //     uc_info_rx->out_snd_device = SND_DEVICE_OUT_SPEAKER_PROTECTED_VBAT;
+    // else
+        // uc_info_rx->out_snd_device = SND_DEVICE_OUT_SPEAKER_PROTECTED;
+    uc_info_rx->out_snd_device = SND_DEVICE_OUT_SPEAKER;
```

Comment:
What even does vbat mean... And do we want the protected speaker here? Neither works...

Reply:
VBAT means exactly what you think it means. VBAT. Battery voltage.

Reply:
What do speakers have to do with battery voltage? Compensation for decreasing voltage and hence volume?

Reply:
Anyway, which of the 3 should we use @kholk?